microsoft/testsuites/core/mana.py

```diff
@@ -0,0 +1,164 @@
+import re
+
+from lisa import (
+    LisaException,
+    Node,
+    SkippedException,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.operating_system import Linux, Posix
+from lisa.sut_orchestrator import AZURE
+from lisa.tools import Dmesg, KernelConfig, Modprobe
+
+
+@TestSuiteMetadata(
+    area="mana_driver",
+    category="functional",
+    description="""
+    This test suite checks for the presence of mana and mana_ib drivers
+    """,
+)
+class ManaDriverCheck(TestSuite):
+    MANA_DRIVER_MESSAGE_PATTERN = re.compile(r"\[\s*[0-9.]+\s*\]\s+mana\s+.*")
+    MANA_IB_DRIVER_MESSAGE_PATTERN = re.compile(r"\[\s*[0-9.]+\s*\]\s+mana_ib\s+.*")
+
+    @TestCaseMetadata(
+        description="""
+        This test case checks for the mana ethernet driver.
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            supported_platform_type=[AZURE], supported_os=[Posix]
+        ),
+    )
+    def verify_mana_en_driver_present(self, node: Node) -> None:
+        if not node.nics.is_mana_device_present():
+            raise SkippedException(
+                "MANA driver tests should be run in a MANA VM environment"
+            )
+
+        distro = node.os
+        if isinstance(distro, Linux):
+            if distro.get_kernel_information().version < "5.15.0":
+                raise SkippedException("MANA is not available for kernels < 5.15")
+
+        mana_driver_name = "mana"
+        dmesg = node.tools[Dmesg]
+        modprobe = node.tools[Modprobe]
+        kconfig = node.tools[KernelConfig]
```

Comment:
we can check config even there is no mana devices

Reply:
added a test just kconfig only